persist/log.go

```diff
@@ -3,28 +3,65 @@ package persist
 import (
 	"log"
 	"os"
+
+	"github.com/NebulousLabs/Sia/build"
 )
 
+// logFileWrapper wraps a log file to perform sanity checks on the Write and
+// Close functions. An error will be returned if Close is called multiple times
+// or if Write is called after Close has been called.
+type logFileWrapper struct {
+	closed bool // Flag is set after Close is called.
+	file   *os.File
+}
```

Comment:
needed to implement the `Write` function.

The alternative was to implement the `Write` function on the type holding the log.Logger, but then when initializing the log.Logger you'd be passing the struct containing it as the interface, and that just seemed wrong.

So instead I opted for a new type on which to implement `Write`. And since I had to do that, I figured it made sense to implement close on it as well, so it could add some safety features. I even considered having `newLogFileWrapper` open the file itself, and still might do that, but I haven't convinced myself that it's the best strategy yet.

But I think it's the right thing to do.